src/validation_workflow/docker/validation_docker.py

```diff
@@ -147,17 +195,41 @@ def run_container(self, OpenSearch_image: str, OpenSearchDashboard_image: str, v
         self.target_yml_file = os.path.join(self.tmp_dir.name, 'docker-compose.yml')
 
         self.version_number = version[0]
-        if self.version_number in self.docker_compose_files:
-            self.source_file = os.path.join('docker', 'release', 'dockercomposefiles', self.docker_compose_files[self.version_number])
-            shutil.copy2(self.source_file, self.target_yml_file)
+        try:
+            source_file = None
+            if self.version_number in self.docker_compose_files:
+                source_file = os.path.join('docker', 'release', 'dockercomposefiles', self.docker_compose_files[self.version_number])
+                shutil.copy2(source_file, self.target_yml_file)
 
-        self.inplace_change(self.target_yml_file, f'opensearchproject/opensearch:{version[0]}', f'{OpenSearch_image}:{version}')
-        self.inplace_change(self.target_yml_file, f'opensearchproject/opensearch-dashboards:{version[0]}', f'{OpenSearchDashboard_image}:{version}')
+            if (self.args.using_staging_artifact_only):
```

Comment:
Use inheritance. This should be refactored into different validation implementations, one for staging and another for otherwise.